aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,336 @@
+  :linenos:
+  :lines: 9-27
+  :emphasize-lines: 6-11 
+  :dedent: 3
+
+The code above demonstrates each of the two factory create methods. 
+
+You can partition your localized strings by controller, area, or have just one container. In the sample app, a dummy class named ``SharedResource`` is used to contain shared resources. Some developers use the ``Startup`` class to contain global or shared strings.  In the sample below, the ``ZebraController`` and the ``SharedResource`` localizers are used:
+
+.. literalinclude:: localization/sample/Controllers/ZebraController.cs
+  :language: c#
+  :linenos:
+  :lines: 9-27
+  :dedent: 3
+
+View localization
+--------------------
+
+The ``IViewLocalizer`` service provides localized strings for a `view <http://docs.asp.net/projects/mvc/en/latest/views/index.html>`_. The ``ViewLocalizer`` class implements this interface and finds the resource location from the view file path. The following code shows how to use the default implementation of ``IViewLocalizer``:
+
+.. literalinclude:: localization/sample/Views/Home/About.cshtml
+  :language: HTML
+  
+The default implementation of ``IViewLocalizer`` finds the resource file based on the view name. There is no option to use a global shared resource file. ``ViewLocalizer`` implements the localizer using ``IHtmlLocalizer``, so Razor doesn't HTML encode the localized string. You can parameterize resource strings and ``IViewLocalizer`` will HTML encode the parameters, but not the resource string. Consider the following Razor markup:
+
+.. code-block:: HTML
+
+  @Localizer["<i>Hello</i> <b>{0}!</b>", UserManager.GetUserName(User)]
+ 
+A French resource file could contain the following:
+
+========================  ===============================  
+Key                       Value    
+========================  ===============================  
+<i>Hello</i> <b>{0}!</b>  <i>Bonjour</i> <b>{0}!</b>  
+========================  ===============================
+
+The rendered view would contain the HTML markup from the resource file.
+
+To use a shared resource file in a view, inject ``IHtmlLocalizer<T>``:
+
+.. literalinclude:: localization/sample/Views/Test/About.cshtml
+  :language: HTML
+  :linenos:
+  :emphasize-lines: 5,12
+
+DataAnnotations localization
+------------------------------------
+
+DataAnnotations error messages are localized with ``IStringLocalizer<Type>``. Using the option ``ResourcesPath = "Resources"``, the error messages in ``RegisterViewModel`` can be stored in either of the following paths:
+
+- Resources/ViewModels.Account.RegisterViewModel.fr.resx
+- Resources/ViewModels/Account/RegisterViewModel.fr.resx
+  
+.. literalinclude:: localization/sample/ViewModels/Account/RegisterViewModel.cs
+  :language: c#
+  :lines: 9-26
+  :dedent: 3
+  
+The run time doesn't look up localized strings for non-validation attributes. In the code above, the "Email" (from ``[Display(Name = "Email")]``) will not be localized.
+
+Configuring localization
```

Comment:
@ryanbrandenburg  I think you're right. I couldn't find a natural place to move it up. However, it does appear as a link in TOC at the top so folks can jump right to it.